### PR TITLE
Venmo Final Amount

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

 -   Add `setIsFinalAmount()` to `VenmoRequest`
     - This removes the text "Subject to change" as seen below

#### Visuals

| Final Amount = false (default) | Final Amount = true |
|-----|-----|
|Screenshot_20240214-084546|Screenshot_20240214-084609|

### Checklist

 - [x] Added a changelog entry
 - [x] Relevant test coverage

### Authors

@jaxdesmarais

### CODE REVIEW

File: Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java [hunk 1]
```diff
@@ -248,6 +249,22 @@ public ArrayList<VenmoLineItem> getLineItems() {
         return lineItems;
     }
 
+    /**
+     * @param finalAmount Optional - Indicates whether the purchase amount is the final amount.
+     *                    Defaults to false.
+     */
+    public void setFinalAmount(boolean finalAmount) {
```
Comment (sarahkoop) [inline Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java:256]: Nitpick - I feel like `setIsFinalAmount` would make it more clear that we are setting a boolean here, not a final amount dollar value - what do you think?

Also, I let's add a public getter for this too for consistency and Kotlin property interop

Comment (jaxdesmarais) [reply Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java]: That totally makes sense, I went back and fourth on it. Updated here: https://github.com/braintree/braintree_android/pull/921/commits/cbdecdc6a88094b13ed7543c474ec628ef68068f

File: Venmo/src/main/java/com/braintreepayments/api/VenmoApi.java [hunk 2]
```diff
@@ -27,6 +27,7 @@ void createPaymentContext(@NonNull final VenmoRequest request, String venmoProfi
             input.put("merchantProfileId", venmoProfileId);
             input.put("customerClient", "MOBILE_APP");
             input.put("intent", "CONTINUE");
+            input.put("isFinalAmount", request.getIsFinalAmountAsString());
```
Comment (sarahkoop) [inline Venmo/src/main/java/com/braintreepayments/api/VenmoApi.java:30]: Does this need to be a string? I think you an just put a boolean into the JSONObject

Comment (jaxdesmarais) [reply Venmo/src/main/java/com/braintreepayments/api/VenmoApi.java:30]: The underlying API needs a string value, same as these two above that are bools on our end but strings for the API:
```java
paysheetDetails.put("collectCustomerShippingAddress", request.getCollectCustomerShippingAddressAsString());
paysheetDetails.put("collectCustomerBillingAddress", request.getCollectCustomerBillingAddressAsString());
```

File: Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java [hunk 3]
```diff
@@ -168,6 +168,64 @@ public void createPaymentContext_whenGraphQLPostError_forwardsErrorToCallback()
         verify(callback).onResult(null, error);
     }
 
+    @Test
+    public void createPaymentContext_withTotalAmountAndSetsFinalAmountToTrue() throws JSONException {
```
Comment (sarahkoop) [inline Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java:172]: Does it work ok if total amount is not set and final amount is set to true? No error just doesn't show total amount at all?

**Review by tdchow — APPROVED** [review]

**Review by agedd — APPROVED** [review]
lgtm! 🚀

File: Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java (hunk 3 above)
Comment (sarahkoop) [reply Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java:172]: The only use case I'd be concerned about is if total amount not set, final amount true (unsure what the behavior would be for that), but as long as it is still ok then, Probably every combination doesn't need a separate unit test

**Review by sarahkoop — APPROVED** [review]

File: Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java (hunk 3 above)
Comment (jaxdesmarais) [reply Venmo/src/test/java/com/braintreepayments/api/VenmoApiUnitTest.java:172]: Yep, total amount is optional for both `VenmoPaymentMethodUsage.SINGLE_USE` and `VenmoPaymentMethodUsage.MULTI_USE`. I just tested both cases with only setting `setIsFinalAmount(true)` without `setTotalAmount` and there are no issues.

File: Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java [hunk 4]
```diff
@@ -248,6 +249,29 @@ public ArrayList<VenmoLineItem> getLineItems() {
...
+        return isFinalAmount;
+    }
+
+    /**
+     * @return Whether or not the purchase amount is the final amount as a string value.
+     */
+    String getIsFinalAmountAsString() {
+        return String.valueOf(this.isFinalAmount);
+    }
+
```
Comment (jaxdesmarais) [inline Venmo/src/main/java/com/braintreepayments/api/VenmoRequest.java:273]:
```suggestion
```

**Pull request merged**